File: Project/Task.java
package Project;
import java.lang.Math;


public class Task {

    private int task_id;
    private int deadline;
    private int duration;
    private double perfect_benefit;

    public Task(int task_id, int deadline, int duration, double perfect_benefit) {
        this.task_id = task_id;
        this.deadline = deadline;
        this.duration = duration;
        this.perfect_benefit = perfect_benefit;
    }

    public int get_task_id() {
        return this.task_id;
    }

    public int get_deadline() {
        return this.deadline;
    }

    public int get_duration() {
        return this.duration;
    }

    public double get_max_benefit() {
        return this.perfect_benefit;
    }

    public double get_late_benefit(int minutes_late) {
        minutes_late = Math.max(0, minutes_late);
        return this.perfect_benefit*Math.exp(-0.0170*minutes_late);
    }
}

File: TimeState/Task.go
package TimeState

import (
"math"
)

type Task struct {
	task_id 	int64
	deadline	int64
	duration	int64
	benefit		float64
}

// Sort Task Interface
type ByID []Task
func (a ByID) Len() int {
	return len(a)
}
func (a ByID) Less(i, j int) bool {
	return a[i].task_id < a[j].task_id
}
func (a ByID) Swap(i, j int) {
	a[i], a[j] = a[j], a[i]
}

func NewTask(task_id int64, deadline int64, duration int64, benefit float64) (Task) {
	var t Task
	t.task_id = task_id
	t.deadline = deadline
	t.duration = duration
	t.benefit = benefit
	return t
}

func (task *Task) GetProfit(curTime int64) float64 {
	return task.benefit * math.Exp(-0.0170 * math.Max(0, float64(curTime + task.duration - task.deadline)))
}
